A32NX/html_ui/JS/A32NX_Util.js

```diff
@@ -46,6 +46,59 @@ A32NX_Util.createMachine = (machineDef) => {
     return machine;
 };
 
+/**
+ * Takes a LatLongAlt or LatLong and returns a vector of spherical co-ordinates
+ * @param {(LatLong | LatLongAlt)} ll
+ */
+A32NX_Util.latLonToSpherical = (ll) => {
+    return [
+        Math.cos(ll.lat * Avionics.Utils.DEG2RAD) * Math.cos(ll.long * Avionics.Utils.DEG2RAD),
+        Math.cos(ll.lat * Avionics.Utils.DEG2RAD) * Math.sin(ll.long * Avionics.Utils.DEG2RAD),
+        Math.sin(ll.lat * Avionics.Utils.DEG2RAD)
+    ];
+};
+
+/**
+ * Computes the intersection point of two bearings on a great circle
+ * @param {(LatLong | LatLongAlt)} latlon1
+ * @param {number} brg1
+ * @param {(LatLong | LatLongAlt)} latlon2
+ * @param {number} brg2
+ */
+A32NX_Util.greatCircleIntersection = (latlon1, brg1, latlon2, brg2) => {
+    // c.f. https://blog.mbedded.ninja/mathematics/geometry/spherical-geometry/finding-the-intersection-of-two-arcs-that-lie-on-a-sphere/
+
+    // TODO should be magvar at the lat, lon
+    const magVar = SimVar.GetSimVarValue("MAGVAR", "degree");
```

Comment:
I had forgotten about this. I think this is the wrong level of abstraction to be dealing with magnetic headings. [Magvar looks to available on navaids from the facility loader since SU3](https://github.com/flybywiresim/a32nx/pull/3868/files#diff-1a4f0f0dbbd73af4b8e2bea0c691cfc053b7d51808c4f3017d31003f39f8db35R298) so I can get the proper mag var now too. I will refactor this a little in addition to the other note from QA.